### PR TITLE
Ben/multi org

### PR DESCRIPTION
**Pull request by blugavere**
Add recursive folder structure reading & labeling support.

https://github.com/cloud-annotations/docusaurus-plugin-openapi/issues/82

image

### CODE REVIEW

File: packages/docusaurus-plugin-openapi/src/openapi/openapi.ts [hunk 1]
```diff
@@ -213,10 +213,12 @@ export async function readOpenapiFiles(
...
-    const sources = await Globby(["**/*.{json,yaml,yml}"], {
-      cwd: openapiPath,
-      ignore: GlobExcludeDefault,
-    });
+    const sources = (
+      await Globby(["**/*.{json,yaml,yml}"], {
+        cwd: openapiPath,
+        ignore: GlobExcludeDefault,
+      })
+    ).filter((x) => !x.includes("_category_")); // todo: regex exclude?
```
Comment (bourdakos1) [inline packages/docusaurus-plugin-openapi/src/openapi/openapi.ts:221]: I think `_category_` should be excluded by `GlobExcludeDefault` above, which excludes anything that starts with an underscore?

File: packages/docusaurus-plugin-openapi/src/openapi/openapi.ts [hunk 2]
```diff
@@ -213,10 +213,12 @@ export async function readOpenapiFiles(
     );
 
     // TODO: Add config for inlcude/ignore
-    const sources = await Globby(["**/*.{json,yaml,yml}"], {
-      cwd: openapiPath,
-      ignore: GlobExcludeDefault,
-    });
+    const sources = (
+      await Globby(["**/*.{json,yaml,yml}"], {
```
Comment (bourdakos1) [inline packages/docusaurus-plugin-openapi/src/openapi/openapi.ts:217]: I prefer to have `await` statements on their own line. For example:
```js
const x = await xxx()
const y = x.map(...).filter(...).sort().etc()
```

**Comment by bourdakos1** [comment]
Can we rename` __test__` to `__fixtures__`?

File: packages/docusaurus-plugin-openapi/src/sidebars/index.ts [hunk 3]
```diff
@@ -59,6 +68,9 @@ export function generateSidebars(items: Item[], options: Options): PropSidebar {
       const info = prototype?.api?.info;
       const fileName = path.basename(source).split(".")[0];
       return {
+        source: prototype?.source,
+        sourceDirName: prototype?.sourceDirName ?? ".",
```
Comment (bourdakos1) [inline packages/docusaurus-plugin-openapi/src/sidebars/index.ts:72]: This makes me a little uncomfortable having the existence of source being tied to the existence of `api.info`. Maybe we should just use `items[0].source` and `items[0].sourceDirName`? However, with hardcoding zero index we have to think if there is ever a case were `items` is an empty array? Maybe an empty directory, or spec that wouldn't generate any pages, etc... We should probably have a case that skips creating the category/subitems if we don't have the right info

**Review by bourdakos1 — CHANGES_REQUESTED** [review]
Looking good, I still need to look over the category tree stuff

File: packages/docusaurus-plugin-openapi/src/openapi/openapi.ts (hunk 1 above)
Comment (blugavere) [reply packages/docusaurus-plugin-openapi/src/openapi/openapi.ts:221]: image

doesn't appear to work

File: packages/docusaurus-plugin-openapi/src/sidebars/index.ts (hunk 3 above)
Comment (blugavere) [reply packages/docusaurus-plugin-openapi/src/sidebars/index.ts:72]: > If there is ever a case were items is an empty array?

I don't think so but can mull on it. 

The current implementation (for all of multi-spec) relies on the folder structure being generated from OpenAPI Specs (& [these lines are only hit if there are > 1 spec files)](https://github.com/blugavere/docusaurus-plugin-openapi/blob/8161387ad207f38dcb80828d919b24f7f7a9a941/packages/docusaurus-plugin-openapi/src/sidebars/index.ts#L85). 

Arbitrary markdown files aren't supported. Introductions are automatically generated from the `info.description` field so in all cases a section is only created for an openapi spec. [Here](https://github.com/blugavere/docusaurus-plugin-openapi/blob/8161387ad207f38dcb80828d919b24f7f7a9a941/packages/docusaurus-plugin-openapi/src/sidebars/index.ts#L66) we're ignoring the auto-generated markdown file from being used to select the sidebar label after being grouped by [source](https://github.com/blugavere/docusaurus-plugin-openapi/blob/8161387ad207f38dcb80828d919b24f7f7a9a941/packages/docusaurus-plugin-openapi/src/sidebars/index.ts#L63).

The fn input items[] is a flattened array of ["auto-generated markdown for spec a", "openapi path for spec a"]. [The .find fn ](https://github.com/blugavere/docusaurus-plugin-openapi/blob/8161387ad207f38dcb80828d919b24f7f7a9a941/packages/docusaurus-plugin-openapi/src/sidebars/index.ts#L65) is getting not the zero index, but the first element in the group which is non-markdown & has the openapi spec info. [If falsy, defaults to filename](https://github.com/blugavere/docusaurus-plugin-openapi/blob/8161387ad207f38dcb80828d919b24f7f7a9a941/packages/docusaurus-plugin-openapi/src/sidebars/index.ts#L77).

File: packages/docusaurus-plugin-openapi/src/openapi/openapi.ts (hunk 1 above)
Comment (bourdakos1) [reply packages/docusaurus-plugin-openapi/src/openapi/openapi.ts]: Ah yea looks like it doesn't ignore json files with `_`
```ts
  // Ignore files starting with _
  '**/_*.{js,jsx,ts,tsx,md,mdx}',

  // Ignore folders starting with _ (including folder content)
  '**/_*/**',

  // Ignore tests
  '**/*.test.{js,jsx,ts,tsx}',
  '**/__tests__/**',
```

Comment (bourdakos1) [reply packages/docusaurus-plugin-openapi/src/openapi/openapi.ts]: maybe try something like this then:
```ts
[...GlobExcludeDefault, '**/_*.{json,yaml,yml}']
```

**Review by bourdakos1 — APPROVED** [review]
Looks good! We can investigate potential issues with an empty array of pages later

**Pull request merged**